### PR TITLE
[hashdb] make the `Hasher` trait useable with `no_std`

### PR DESCRIPTION
**Pull request by dvdplm**
Enable use of the `Hasher` trait in a `no_std` context.

### CODE REVIEW

**Comment by ordian** [comment]
Can we make sure all these crates compile fine with `--no-default-features` in CI?

File: .travis.yml [hunk 1]
```diff
@@ -8,4 +8,6 @@ matrix:
 script:
   - cargo build
   - cargo test --all --exclude uint
-  - cd uint/ && cargo test --features=std,impl_quickcheck_arbitrary
+  - cd uint/ && cargo test --features=std,impl_quickcheck_arbitrary && cd ..
+  - cd hashdb/ && cargo test --no-default-features && cd ..
```
Comment (ordian) [inline .travis.yml]: Do we need to pass these crates to `--exclude` above, or are we going to have some `std/no-std`-specific tests in the future?

Comment (dvdplm) [reply .travis.yml]: At the moment `hashdb` only defines traits and have no tests, so `cargo test` amounts to essentially typechecking. `plain_hasher` has one test so this runs it both ways (std/no_std). 
I think this is ok for now.

**Comment by dvdplm** [comment]
> moving Hasher to a separate crate or module

I agree with both of you. I filed [an issue](https://github.com/paritytech/parity-common/issues/23)

**Pull request merged**